server/src/services/cronSchedule/cronSchedule.js

```diff
@@ -35,6 +35,29 @@ const getEmbedIDDocVStep = (steps) => {
   });
 };
 
+/**
+ * // TODO: Document what this is doing and why (including clarifying what the unusual txId2 property is). And refactor / rename this function to be more appropriate.
+ *
+ * @param {*} truliooInstance
+ * @param {*} applicant
+ * @param {*} step
+ */
+const ifStepHasIdsSaveApplicantAndCreateTransaction = async (truliooInstance, applicant, step) => {
+  const { transactionId, transactionRecordId } = step;
+
+  if (transactionId && transactionRecordId) {
+    applicant.txId2 = transactionId; // Before splitting out this function, this line had: error  Possible race condition: `applicant.txId2` might be assigned based on an outdated state of `applicant`  require-atomic-updates
+    await applicant.save();
+
```

Comment:
Lines:
    applicant.txId2 = transactionId; // Before splitting out this function, this line had: error  Possible race condition: `applicant.txId2` might be assigned based on an outdated state of `applicant`  require-atomic-updates
    await applicant.save();
better to replace with:
    await Applicant.findOneAndUpdate({id: applicant._id}, {txId2: transactionId});